File: SurmisingASprintersSpeed.java
import java.io.IOException;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        List<Observation> observations1 = new ArrayList<>();
        Scanner scanner = new Scanner(System.in);
        int n = Integer.parseInt(scanner.nextLine());
        for (int i = 0; i < n; i++) {
            String[] tokens = scanner.nextLine().split("\\s");
            observations1.add(new Observation(Integer.parseInt(tokens[0]), Integer.parseInt(tokens[1])));
        }


        scanner.close();

        observations1.sort(Comparator.comparing(o -> o.time));

        int timeDiff = 0;
        int meterDiff = 0;
        double temp = 0;
        double maxDivided = 0;
        for (int i = 0; i < observations1.size() - 1; i++) {
            timeDiff = observations1.get(i + 1).time - observations1.get(i).time;

            meterDiff = Math.abs(observations1.get(i + 1).spot - observations1.get(i).spot);

            temp = 1.0 * meterDiff / timeDiff;

            if (temp > maxDivided) {
                maxDivided = temp;
            }
        }


        System.out.println(maxDivided);
    }


    static class Observation {
        public int time;
        public int spot;

        public Observation(int time, int spot) {
            this.time = time;
            this.spot = spot;
        }


    }


}

File: SurmisingASprintersSpeed/SurmisingASprintersSpeed.py
class Observation:
    def __init__(self, time, spot):
        self.time = time
        self.spot = spot


n = int(input())
count = n
observations1 = []
while count > 0:
    s = input().split(" ")
    observations1.append(Observation(int(s[0]), int(s[1])))
    count -= 1
observations1.sort(key=lambda x: x.time)
time_diff = 0
meter_diff = 0
temp = 0.0
max_divided = 0.0
for i in range(len(observations1) - 1):
    time_diff = observations1[i+1].time - observations1[i].time
    meter_diff = abs(observations1[i+1].spot - observations1[i].spot)
    temp = 1.0 * meter_diff / time_diff
    if temp > max_divided:
        max_divided = temp
print(max_divided)